elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -755,6 +756,27 @@ public boolean isIdGenerated() {
         return getIdAnnotations().stream().anyMatch(a -> a.annotationType().equals(GeneratedValue.class));
     }
 
+    /**
+     * Indicates if the ID is mapped to another entity or not.
+     *
+     * @return Boolean
+     */
+    public boolean isIdMapped() {
+        final Class<?> cls = getObject().getClass();
+        for (final AccessibleObject obj : dictionary.getEntityBinding(cls).getAllFields()) {
```

Comment:
Rather than walking the attributes this during the request processing, I think this logic could move to the `EntityBInding` class.  The binding can store properties about the id (if it is generated or mapped) - and all of this would be computed once during server init.

The entity binding walks each field or relationship when an entity is bound.  We could check for annotations like GeneratedValue and MapsId there.

Reply:
make sense 👍